modules/apps/notifications/notifications-web/src/main/java/com/liferay/notifications/web/internal/portlet/configuration/icon/DeliveryPortletConfigurationIcon.java

```diff
@@ -59,10 +59,9 @@ public String getMethod() {
 	public String getOnClick(
 		PortletRequest portletRequest, PortletResponse portletResponse) {
 
-		StringBundler sb = new StringBundler(12);
+		StringBundler sb = new StringBundler(11);
 
-		sb.append("Liferay.Portlet.openModal({iframeBodyCssClass: ");
-		sb.append("'dialog-with-footer', namespace: '");
+		sb.append("Liferay.Portlet.openModal({namespace: '");
```

Comment:
You kept the `props.iframeBodyCssClass` in the API. Why remove it from here?

Reply:
I kept it in API because there are rare places where we add a custom class, like [here](https://github.com/liferay/liferay-portal/blob/182043585bfd9dde2dd0ddab7ed0fdb8760dc758/modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/page.jsp#L103). But, thinking about it now, we should remove it completely from API since it's not reliable (navigation resets it). Usages will need to be modified to use internal class on their main content, but that should not be a problem, and will result be a cleaner solution overall.

Reply:
Based on https://github.com/john-co/liferay-portal/pull/293#pullrequestreview-408944148, I'd just keep it both in the API and in here... and then once it's merged revisit this to see if there's another approach. I know you already covered several roads, but I'd like to give this a look.

Reply:
@jbalsas After most of the day thinking about this:
- It does not make sense to revert **only** setting of `dialog-with-footer` in usages. This will not do anything, as we are always setting that class in theme. The only effect will be duplicate css classes on elements.
- If we revert **both** usages and setting in theme, we will end up with breaking bugs. For example, you cannot set web content in the web content display app. If the class change in theme is unacceptable, we cannot push this task forward until we find a completely different solution.
- The bug I mentioned in https://github.com/john-co/liferay-portal/pull/293#discussion_r422965797 is minor, basically we have an unnecessary footer-height bottom padding in some modals. It would be reasonably safe to push this as it is, with a followup fix, or approach rewrite.
- I found a way to fix the bug, but again, I am not sure if the approach is acceptable, as it further impacts theme. It consists of removing `.dialog-with-footer` from theme entirely, and rely on `.dialog-footer` class. `.dialog-footer` styles are on footer container, that is always present and set in iframe content.